### DISCUSSION TITLE
Integration with external RAG pipelines

### DISCUSSION THREAD

**Discussion by anmolbhatia05** [open-webui/open-webui#3973] [General]
**Is your feature request related to a problem? Please describe.**
Our company has a lot of internal documents stored in a vector DB, system prompt that works well with our retrieval use cases and return prompt templates, we would like to use open-webui as a frontend to this RAG system. 

**Describe the solution you'd like**
It would make this tool really extensible if there is a way to integrate existing RAG pipelines that have their own embeddings store, system prompts, prompt templates, etc. 

**Describe alternatives you've considered**
We made our own frontend, but open-webui has so many other functionalities that we would like to somehow integrate these. 

**Additional context**

**Comment by tjbck** [comment]
https://github.com/open-webui/open-webui/discussions/3687#discussioncomment-10016728
